Extension method on Java class fails in runtime

## Compiler version

3.0.0

## Minimized code

```Scala
String.nn
```

## Output (click arrow to expand)
<details>

```scala
Exception in thread "main" java.lang.NoClassDefFoundError: java/lang/String$
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.getDeclaredMethods(Class.java:1975)
	at dotty.tools.repl.Rendering.valueOf(Rendering.scala:102)
	at dotty.tools.repl.Rendering.renderVal(Rendering.scala:141)
	at dotty.tools.repl.ReplDriver.$anonfun$13(ReplDriver.scala:323)
	at scala.collection.immutable.List.flatMap(List.scala:293)
	at scala.collection.immutable.List.flatMap(List.scala:79)
	at dotty.tools.repl.ReplDriver.extractAndFormatMembers$1(ReplDriver.scala:323)
	at dotty.tools.repl.ReplDriver.renderDefinitions$$anonfun$2(ReplDriver.scala:346)
	at scala.Option.map(Option.scala:242)
	at dotty.tools.repl.ReplDriver.renderDefinitions(ReplDriver.scala:349)
	at dotty.tools.repl.ReplDriver.compile$$anonfun$2(ReplDriver.scala:268)
	at scala.util.Either.fold(Either.scala:189)
	at dotty.tools.repl.ReplDriver.compile(ReplDriver.scala:285)
	at dotty.tools.repl.ReplDriver.interpret(ReplDriver.scala:212)
	at dotty.tools.repl.ReplDriver.loop$1(ReplDriver.scala:145)
	at dotty.tools.repl.ReplDriver.runUntilQuit$$anonfun$1(ReplDriver.scala:148)
	at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:167)
	at dotty.tools.repl.ReplDriver.runUntilQuit(ReplDriver.scala:148)
	at dotty.tools.repl.ReplDriver.tryRunning(ReplDriver.scala:114)
	at dotty.tools.repl.Main$.main(Main.scala:6)
	at dotty.tools.repl.Main.main(Main.scala)
Caused by: java.lang.ClassNotFoundException: java.lang.String$
	at dotty.tools.repl.AbstractFileClassLoader.findClass$$anonfun$1(AbstractFileClassLoader.scala:44)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at dotty.tools.repl.AbstractFileClassLoader.findClass(AbstractFileClassLoader.scala:46)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at dotty.tools.repl.AbstractFileClassLoader.loadClass(AbstractFileClassLoader.scala:55)
	... 23 more
```
</details>

## Expectation

This should not throw an exception.

Minimized (without inline):

```scala
extension [T](x: T) def ext: T = x
def foo = String.ext
```

FWIW `-Ycheck:all` catches the annomaly after erasure:

```scala
*** error while checking /home/olivier/workspace/dotty/tests/run/12537.scala after phase erasure ***                         
exception occurred while compiling /home/olivier/workspace/dotty/tests/run/12537.scala                                       
Exception in thread "main" java.lang.AssertionError: assertion failed: Java class can't be used as value: Ident(String)
```

--- a/compiler/src/dotty/tools/dotc/transform/SymUtils.scala
+++ b/compiler/src/dotty/tools/dotc/transform/SymUtils.scala
@@ -64,6 +64,8 @@ object SymUtils:
 
     def isSuperAccessor(using Context): Boolean = self.name.is(SuperAccessorName)
 
+    def isNoValue(using Context): Boolean = self.is(Package) || self.isAllOf(JavaModule)
+
     /** Is this a type or term parameter or a term parameter accessor? */
     def isParamOrAccessor(using Context): Boolean =
       self.is(Param) || self.is(ParamAccessor)

--- a/compiler/src/dotty/tools/dotc/typer/Checking.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Checking.scala
@@ -713,7 +713,7 @@ object Checking {
 
   def checkValue(tree: Tree)(using Context): Unit =
     val sym = tree.tpe.termSymbol
-    if sym.is(Flags.Package) || sym.isAllOf(Flags.JavaModule) && !ctx.isJava then
+    if sym.isNoValue && !ctx.isJava then
       report.error(JavaSymbolIsNotAValue(sym), tree.srcPos)
 
   def checkValue(tree: Tree, proto: Type)(using Context): tree.type =

--- a/compiler/src/dotty/tools/dotc/typer/Typer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Typer.scala
@@ -3245,6 +3245,8 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
       rememberSearchFailure(qual,
         SearchFailure(qual.withType(NestedFailure(ex.toMessage, selectionProto))))
 
+    if qual.symbol.isNoValue then return EmptyTree
+
     // try an extension method in scope
     try
       val nestedCtx = ctx.fresh.setNewTyperState()

--- a/tests/neg/i12537.scala
+++ b/tests/neg/i12537.scala
@@ -1,0 +1,2 @@
+extension [T](x: T) def ext: T = x
+def foo = String.ext  // error

--- a/tests/neg/i7750.scala
+++ b/tests/neg/i7750.scala
@@ -1,3 +1,3 @@
-package object A extends runtime.A {
+package object A extends runtime.A {  // error
     implicit def a( : ) = 1() // error // error // error
 }

